### PR TITLE
Allow to use ocaml-lsp-server with opam pin / esy link workflows

### PR DESCRIPTION
**Pull request by andreypopp**
This is a workaround for ocaml/dune#2982 which allows to use ocaml-lsp with opam
pin and esy link workflows.

Summary:

- Remove a dep on lsp package as we treat lsp as a part of the vendored
  set of deps.

- Fix build command first to consider the entire workspace while
  building ocaml-lsp-server package. This is needed so we see vendored
  merlin and other packages.

This also removes `with-test` and `with-doc` which can be restored later once we
address that in dune.

### CODE REVIEW

**Comment by andreypopp** [comment]
This should land after #42.

File: dune-project [hunk 1]
```diff
@@ -15,30 +15,4 @@
 
 (source (github ocaml/ocaml-lsp))
 
-(generate_opam_files true)
```
Comment (andreypopp) [inline dune-project]: Had to remove generation of opam files for now to be able to provide custom build commands.

File: ocaml-lsp-server.opam [hunk 2]
```diff
@@ -14,10 +13,9 @@ license: "ISC"
 homepage: "https://github.com/ocaml/ocaml-lsp"
 bug-reports: "https://github.com/ocaml/ocaml-lsp/issues"
 depends: [
-  "result"
-  "lsp"
   "yojson"
   "ppx_yojson_conv_lib"
+  "ocamlfind" {>= "1.5.2"}
```
Comment (andreypopp) [inline ocaml-lsp-server.opam]: This is needed as merlin depends on `findlib` lib.

File: ocaml-lsp-server.opam [hunk 3]
```diff
@@ -26,13 +24,20 @@ build: [
...
+    "--ignore-promoted-rules"
+    "--no-config"
+    "--profile"
+    "release"
+    "%{name}%.install"
+  ]
+  [
+    "cp"
+    "%{_:build}%/_build/default/%{name}%.install"
+    "%{name}%.install"
```
Comment (andreypopp) [inline ocaml-lsp-server.opam]: This is a sad hack to make it work both with dune 1.0 an dune 2.0 (in dune 2.0 `--ignore-promoted-rules` doesn't promote install files but `--promote-install-files` is only available in dune 2.0).

I think it's worth it.

File: ocaml-lsp-server.opam [hunk 4]
```diff
@@ -26,13 +24,20 @@ build: [
...
-    "@runtest" {with-test}
-    "@doc" {with-doc}
+    "--root"
+    "."
+    "--ignore-promoted-rules"
+    "--no-config"
+    "--profile"
+    "release"
+    "%{name}%.install"
+  ]
```
Comment (andreypopp) [inline ocaml-lsp-server.opam]: My intention with this is that this invocation is the same as `-p %{name}%` but allows to see other packages in the workspace.

File: ocaml-lsp-server.opam (hunk 3 above)
Comment (rgrinberg) [reply ocaml-lsp-server.opam]: What is `%{_:build}%`?

File: dune-project (hunk 1 above)
Comment (rgrinberg) [reply dune-project]: There's no need to remove it. It's possible to override the build step only in the generated .opam file. See how we do it in dune for example https://github.com/ocaml/dune/blob/master/dune.opam.template

File: ocaml-lsp-server.opam (hunk 3 above)
Comment (andreypopp) [reply ocaml-lsp-server.opam]: Build dir of the package opam is building currently.

File: dune-project (hunk 1 above)
Comment (andreypopp) [reply dune-project]: Whoa, didn't know dune supports it. Fixed.

**Pull request merged**